### PR TITLE
docs: fix `useActionData`/`useLoaderData` usage

### LINKED ISSUE #4110: add SerializeFrom to Developer Blog Tutorial
### What version of Remix are you using?

1.7.0

### Steps to Reproduce

I followed the getting started instructions and began the [Developer Blog Tutorial](https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md).

### Expected Behavior

I expected to not have type errors while following the tutorial.

### Actual Behavior

In the "A little refactoring" section, the return value of `useLoaderData` has a type error of "Type 'string' is not assignable to type 'Date'", because of a mismatch between `Date` and `string` types generated by Prisma and the serialized type of the `createdAt` and `updatedAt` properties.

I did a little digging and found the `SerializeFrom` type that is new to `1.7.0`. I was able to fix the type error using the following code. Though because there are no examples of this, I have no idea if this is the proper way to type LoaderData.

```ts
import type {SerializeFrom } from "@remix-run/node";
import { json } from "@remix-run/node";
import { Link, useLoaderData } from "@remix-run/react";

import { getPosts } from "~/models/post.server";

export const loader = async () => {
  return json({
    posts: await getPosts(),
  });
};

type LoaderData = SerializeFrom<typeof loader>;

export default function Posts() {
  const { posts } = useLoaderData() as LoaderData;
  return (
    <main>
      <h1>Posts</h1>
      <ul>
        {posts.map((post) => (
          <li key={post.slug}>
            <Link to={post.slug} className="text-blue-600 underline">
              {post.title}
            </Link>
          </li>
        ))}
      </ul>
    </main>
  );
}
```

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
See https://github.com/remix-run/remix/releases/tag/remix@1.6.5

---

Closes #4110
Closes #4115
Closes #4609

Just like I did in https://github.com/remix-run/examples/pull/82, https://github.com/remix-run/examples/pull/83, https://github.com/remix-run/examples/pull/84 & https://github.com/remix-run/examples/pull/85

### CODE REVIEW

File: docs/tutorials/jokes.md [hunk 1]
```diff
@@ -4285,7 +4227,7 @@ export function ErrorBoundary({ error }: { error: Error }) {
 <summary>app/routes/jokes/$jokeId.tsx</summary>
 
 ```tsx filename=app/routes/jokes/$jokeId.tsx lines=[6,21-25,42-53]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:4229]: Need to update highlighted lines

File: docs/tutorials/jokes.md [hunk 2]
```diff
@@ -4354,7 +4291,6 @@ export function ErrorBoundary() {
 <summary>app/routes/jokes/index.tsx</summary>
 
 ```tsx filename=app/routes/jokes/index.tsx lines=[6,21-25,44-57]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:4293]: Need to update highlighted lines.

File: docs/tutorials/jokes.md [hunk 3]
```diff
@@ -4653,8 +4574,8 @@ And then the `action` can determine whether the intention is to delete based on
 ```tsx filename=app/routes/jokes/$jokeId.tsx lines=[6,15,34-64,74-83,92-98,106-112]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:4574]: Need to update highlighted lines.

File: docs/tutorials/jokes.md [hunk 4]
```diff
@@ -4789,8 +4705,8 @@ Now that people will get a proper error message if they try to delete a joke tha
 ```tsx filename=app/routes/jokes/$jokeId.tsx lines=[16,20,26,37,82-93]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:4705]: Need to update highlighted lines.

File: docs/tutorials/jokes.md [hunk 5]
```diff
@@ -5375,8 +5278,8 @@ export default function Login() {
 
 ```tsx filename=app/routes/jokes/$jokeId.tsx lines=[4,21-36]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:5279]: Need to update highlighted lines.

File: docs/tutorials/jokes.md [hunk 6]
```diff
@@ -5863,8 +5759,8 @@ export function JokeDisplay({
 
 ```tsx filename=app/routes/jokes/$jokeId.tsx lines=[19,97]
```
Comment (pcattori) [inline docs/tutorials/jokes.md:5760]: Need to update highlighted lines.

**Review by pcattori — COMMENTED** [review]
Thanks for taking this on! Docs updates can be tedious and we really appreciate it 🙏

It's looking quite good! Just left a couple comments mostly about updating highlighted lines in code examples.

**Comment by MichaelDeBoey** [comment]
Thanks for the thorough review @pcattori! 🙏

Fixed all the remarks & some minor other things as well, so should be good to be merged now.

**Review by pcattori — APPROVED** [review]

**Comment by petetnt** [comment]
Hi @MichaelDeBoey, this PR causes issues (or leaves them unfixed?) at least with `docs/guides/data-writes.md` where setting `defaultValue`s as per the documentation throws the following type error (see https://remix.run/docs/en/v1/guides/data-writes#graduate-to-form-and-add-pending-ui):

```
Type 'string | SerializeObject<UndefinedToOptional<File>> | undefined' is not assignable to type 'string | number | readonly string[] | undefined'.
  Type 'SerializeObject<UndefinedToOptional<File>>' is not assignable to type 'string | number | readonly string[] | undefined'.ts(2322)
index.d.ts(1851, 9): The expected type comes from property 'defaultValue' which is declared here on type 'DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'
```

**Pull request merged**